### ISSUE TITLE
Custom Dataset Validation

### ISSUE THREAD

**Issue by leftbackn3** [Daniil-Osokin/lightweight-human-pose-estimation.pytorch#274]
Hi, I referred to a previously closed issue and created my custom dataset in accordance with the specifications.

However when i am running the validation code. I am receiving null values for pose entries and coco keypoints. Besides avg paf and avg heatmap matrices are not null. Can you please suggest me a possible way to overcome this ?

I am continuing training on the existing pretrained model since I have a small dataset.

**Comment by leftbackn3** [comment]
keypoint = [other_annotation['keypoints'][i * 3], other_annotation['keypoints'][i * 3 + 1], 2]
                    if other_annotation['keypoints'][i * 3 + 2] == 1:
                        keypoint[2] = 0
                    elif other_annotation['keypoints'][i * 3 + 2] == 2:
                        keypoint[2] = 1

I had a made a change in this part of prepare_train_labels. So while creating the annotations, if a keypoint is absent in the frame, I had marked that as [0,0,0] and if its present then(x,y,1)
so i created another file prepare_custom_train_labels:
where i had made this change-
 keypoint = [other_annotation['keypoints'][i * 3], other_annotation['keypoints'][i * 3 + 1], 2]
                    if annotation['keypoints'][i * 3 + 2] == 1:
                    keypoint[2] = 1
                elif annotation['keypoints'][i * 3 + 2] == 0:
                    keypoint[2] = 2

I have not taken occlusion into consideration

**Comment by Daniil-Osokin** [comment]
Hi! It is hard to say what went wrong based on this description. Did you training loss decreased significantly? What is visualization of heatmaps on validation dataset, did they contain peaks in the place of keypoints?

**Comment by leftbackn3** [comment]
Hi, I have resolved this issue. The real problem was the size of the dataset.
